Modules/Loadable/Texts/Widgets/qMRMLTextWidget.h

```diff
@@ -0,0 +1,131 @@
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+  This file was originally developed by Kyle Sunderland, PerkLab, Queen's University
+  and was supported through CANARIE's Research Software Program, and Cancer
+  Care Ontario.
+
+==============================================================================*/
+
+#ifndef __qMRMLTextWidget_h
+#define __qMRMLTextWidget_h
+
+// SlicerQt includes
+#include "qSlicerWidget.h"
+
+// Text widgets includes
+#include "qSlicerTextsModuleWidgetsExport.h"
+#include "ui_qMRMLTextWidget.h"
+
+class vtkMRMLNode;
+class vtkMRMLTextNode;
+class qMRMLTextWidgetPrivate;
+
+/// \ingroup Slicer_QtModules_Texts
+class Q_SLICER_MODULE_TEXTS_WIDGETS_EXPORT qMRMLTextWidget : public qSlicerWidget
+{
+  Q_OBJECT
+
+public:
+  typedef qSlicerWidget Superclass;
+  qMRMLTextWidget(QWidget *parent=nullptr);
+  ~qMRMLTextWidget() override;
+
+  Q_PROPERTY(bool readOnly READ readOnly WRITE setReadOnly)
+  Q_PROPERTY(bool continuousUpdate READ continuousUpdate WRITE setContinuousUpdate)
+
+  /// Get the table node the columns are edited of.
+  Q_INVOKABLE vtkMRMLTextNode* mrmlTextNode() const;
+  Q_INVOKABLE vtkMRMLNode* mrmlNode() const;
+
+  /// Returns true if the text editor is read only
+  bool readOnly();
+
+  /// Returns true if the text editor is continuously updated from the text node and vice versa
+  bool continuousUpdate();
+
+public slots:
+
+  /// Reimplemented from qSlicerWidget
+  void setMRMLScene(vtkMRMLScene* scene) override;
+
+  /// Set the read only property of the text editor.
+  /// \sa readOnly()
+  void setReadOnly(bool readOnly);
+
+  /// Get the continuous update propery of the text editor
+  /// If true, the text editor will propogate the text to the vtkMRMLTextNode as it is modified, and vice versa.
+  /// If false, the text editor will only update the node when "Save" is clicked, and changes from the vtkMRMLTextNode will not be propogated
+  /// if the text is being edited.
+  /// \sa continuousUpdate()
+  void setContinuousUpdate(bool continuousUpdate);
+
+  /// Set the currently observed text node
+  /// \sa mrmlTextNode()
+  void setMRMLTextNode(vtkMRMLTextNode* textNode);
+
+  /// Utility function to simply connect signals/slots with Qt Designer
+  /// \sa mrmlNode()
+  void setMRMLNode(vtkMRMLNode* textNode);
+
+protected slots:
+  /// Update the GUI to reflect the currently selected text node.
+  void updateWidget();
+
+  /// Method invoked when the contents of the text node is modified
+  void onTextNodeModified();
+
+  /// Method invoked when the contents of the text edit is changed
+  void onTextChanged();
+
+  /// Method invoked when the "Edit" button is clicked
+  void onEditClicked();
```

Comment:
onEditClicked -> onEditButtonClicked ? Same comment for Cancel and Save

That said, I wonder if it makes sense to have these signals. Having signal like `updateRequested` and `updateCanceled` would be more generic  and would also nicely complement the existing `updateFinished` signal

Reply:
Changed methods to XButtonClicked(). Added new signals (updateRequested, editingChanged(bool)).